docs/src/main/sphinx/security/file-system-access-control.rst

```diff
@@ -331,6 +359,100 @@ The example below defines the following table access policy:
       ]
     }
 
+.. _system-file-function-rules:
+
+Function rules
+^^^^^^^^^^^^^^
+
+These rules control the user's ability to execute SQL functions,
+such as :doc:`aggregate functions </functions/aggregate>`, scalar functions,
+:doc:`table functions </functions/table>` and :doc:`window functions </functions/window>`.
+
+Each function rule is composed of the following fields:
+
+* ``user`` (optional): regular expression to match against user name.
+  Defaults to ``.*``.
+* ``role`` (optional): regular expression to match against role names.
+  Defaults to ``.*``.
+* ``group`` (optional): regular expression to match against group names.
+  Defaults to ``.*``.
+* ``catalog`` (optional): regular expression to match against catalog name.
+  Defaults to ``.*``.
+* ``schema`` (optional): regular expression to match against schema name.
+  Defaults to ``.*``.
+* ``function`` (optional): regular expression to match against function names.
+  Defaults to ``.*``.
+* ``privileges`` (required): zero or more of ``EXECUTE``, ``GRANT_EXECUTE``.
+* ``function_kinds`` (required): one or more of ``AGGREGATE``, ``SCALAR``,
+  ``TABLES``, ``WINDOW``. When a user defines a rule for ``AGGREGATE``, ``SCALAR``
+  or ``WINDOW`` functions, the ``catalog`` and ``schema`` fields are disallowed.
+
+
+To deny all :doc:`table functions </functions/table>` from any catalog,
+use the following rules:
+
+.. code-block:: json
+
+    {
+      "functions": [
+        {
+          "privileges": [
+            "EXECUTE",
+            "GRANT_EXECUTE"
+          ],
+          "function_kinds": [
+            "SCALAR",
+            "AGGREGATE",
+            "WINDOW"
+          ]
+        },
+        {
+          "privileges": [],
+          "function_kinds": [
+            "TABLE"
+          ]
+        }
+      ]
+    }
+
+
+The following example allows the ``admin`` user to execute ``query`` table
+function from any catalog:
+
+.. note::
+
+    It is recommended to only allow admins to use the ``query``
```

Comment:
Do not make this recommendation.. change to normal paragraph that says the following section disallows usage of the query table function. And maybe add a reason why that might make sense (query passthrough ignores access controls)

Reply:
Got it, rewritten.